### PR TITLE
TF models for R3

### PR DESCRIPTION
**Pull request by jkamelin**

### CODE REVIEW

File: models/public/mask_rcnn_inception_v2_coco/mask_rcnn_inception_v2_coco.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ bounding boxes. For each detection, the description has the format:
     - `conf` - confidence for the predicted class
     - (`x_min`, `y_min`) - coordinates of the top left bounding box corner (coordinates stored in normalized format, in range [0, 1])
     - (`x_max`, `y_max`) - coordinates of the bottom right bounding box corner  (coordinates stored in normalized format, in range [0, 1])
-2. name: `masks`
+2. Segmentation mask, name: `masks`, shape: [N, 90, 33, 33], where N is the number of detected masks.
```
Comment (eaidova) [inline models/public/mask_rcnn_inception_v2_coco/mask_rcnn_inception_v2_coco.md]: is mask size 33 x 33?
As I remember the standard size is 28 x 28

File: models/public/mask_rcnn_inception_resnet_v2_atrous_coco/mask_rcnn_inception_resnet_v2_atrous_coco.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+| MParams                         | 92.368                                    |
+| Source framework                | Tensorflow\*                              |
+
+## Performance
+
+## Input
+
+### Original Model
+
+Image, name: `image_tensor`, shape: [1x800x800x3], format: [BxHxWxC],
```
Comment (eaidova) [inline models/public/mask_rcnn_inception_resnet_v2_atrous_coco/mask_rcnn_inception_resnet_v2_atrous_coco.md]: I'm not sure about size specification, in TF model has dynamic shape and image scaled with aspect ratio saving (min size is 800, max size is 1365)

File: models/public/mask_rcnn_inception_resnet_v2_atrous_coco/mask_rcnn_inception_resnet_v2_atrous_coco.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+   where:
+
+    - B - batch size
+    - C - vector of 3 values in format [H,W,S], where H is an image height, W is an image width, S is an image scale factor (usually 1)
+
+## Output
+
+### Original Model
+
+1. Classifier, name: `detection_classes`. Contains predicted bounding boxes classes in a range [1, 91]. The model was trained on the Microsoft\* COCO dataset version with 90 categories of objects.
```
Comment (eaidova) [inline models/public/mask_rcnn_inception_resnet_v2_atrous_coco/mask_rcnn_inception_resnet_v2_atrous_coco.md]: I think a notice that 0 is background also important to add.

File: models/public/mask_rcnn_inception_v2_coco/mask_rcnn_inception_v2_coco.md (hunk 1 above)
Comment (jkamelin) [reply models/public/mask_rcnn_inception_v2_coco/mask_rcnn_inception_v2_coco.md]: This is what i got after conversion

File: tools/downloader/list_topologies.yml [hunk 4]
```diff
@@ -385,6 +385,164 @@ topologies:
...
+        sha256: 816afc8987c2152cbd0e24af3a6fb82086cea0645a38778884bd166c0d016bb5
+        source: http://download.tensorflow.org/models/object_detection/mask_rcnn_inception_resnet_v2_atrous_coco_2018_01_28.tar.gz
+    output: "instance_segmentation/common/mask_rcnn/mask_rcnn_inception_resnet_v2_atrous_coco/tf"
+    postprocessing:
+      - $type: unpack_archive
+        format: gztar
+        file: mask_rcnn_inception_resnet_v2_atrous_coco_2018_01_28.tar.gz
+    model_optimizer_args:
+      - --framework=tf
+      - --data_type=FP16
```
Comment (eaidova) [inline tools/downloader/list_topologies.yml]: Why do you set for all new instance segmentation models and detection models FP16 data type and for mobilenet FP32? is this done intentionally?

File: models/public/mask_rcnn_inception_resnet_v2_atrous_coco/mask_rcnn_inception_resnet_v2_atrous_coco.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+bounding boxes. For each detection, the description has the format:
+[`image_id`, `label`, `conf`, `x_min`, `y_min`, `x_max`, `y_max`],
+    where:
+
+    - `image_id` - ID of the image in the batch
+    - `label` - predicted class ID
+    - `conf` - confidence for the predicted class
+    - (`x_min`, `y_min`) - coordinates of the top left bounding box corner (coordinates stored in normalized format, in range [0, 1])
+    - (`x_max`, `y_max`) - coordinates of the bottom right bounding box corner  (coordinates stored in normalized format, in range [0, 1])
+2. Segmentation mask, name: `masks`, shape: [N, 90, 33, 33], where N is the number of detected masks.
```
Comment (eaidova) [inline models/public/mask_rcnn_inception_resnet_v2_atrous_coco/mask_rcnn_inception_resnet_v2_atrous_coco.md]: I think it is important to explain why 90 (not 91) that background class excluded from mask calculation.
And more correct description is segmentation heatmaps for all classes for every output bounding box

File: models/public/mask_rcnn_inception_v2_coco/mask_rcnn_inception_v2_coco.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+bounding boxes. For each detection, the description has the format:
+[`image_id`, `label`, `conf`, `x_min`, `y_min`, `x_max`, `y_max`],
+    where:
+
+    - `image_id` - ID of the image in the batch
+    - `label` - predicted class ID
+    - `conf` - confidence for the predicted class
+    - (`x_min`, `y_min`) - coordinates of the top left bounding box corner (coordinates stored in normalized format, in range [0, 1])
+    - (`x_max`, `y_max`) - coordinates of the bottom right bounding box corner  (coordinates stored in normalized format, in range [0, 1])
+2. Segmentation mask, name: `masks`, shape: [N, 90, 33, 33], where N is the number of detected masks.
```
Comment (eaidova) [inline models/public/mask_rcnn_inception_v2_coco/mask_rcnn_inception_v2_coco.md]: I observe IR, in this model `mask` layer has shape [N, 90, 15, 15]

File: models/public/mask_rcnn_inception_v2_coco/mask_rcnn_inception_v2_coco.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+1. Classifier, name: `detection_classes`. Contains predicted bounding boxes classes in a range [1, 91]. The model was trained on the Microsoft\* COCO dataset version with 90 categories of objects, 0 class is for background.
+2. Probability, name: `detection_scores`. Contains probability of detected bounding boxes.
+3. Detection box, name: `detection_boxes`. Contains detection boxes coordinates in a format `[y_min, x_min, y_max, x_max]`, where (`x_min`, `y_min`)  are coordinates of the top left corner, (`x_max`, `y_max`) are coordinates of the right bottom corner. Coordinates are rescaled to input image size.
+4. Detections number, name: `num_detections`. Contains the number of predicted detection boxes.
+5. Segmentation mask, name: `detection_masks`. Contains segmentation masks of detected objects for each class.
+
+### Converted Model
+
+1. The array of summary detection information, name: `reshape_do_2d`, shape: [1, 1, N, 7], where N is the number of detected
```
Comment (eaidova) [inline models/public/mask_rcnn_inception_v2_coco/mask_rcnn_inception_v2_coco.md]: why [1, 1, N, 7]? `reshape_do_2d` should get [1, 1, N, 7] as input and return [N, 7] as output. max N is 100.

File: models/public/mask_rcnn_inception_v2_coco/mask_rcnn_inception_v2_coco.md (hunk 6 above)
Comment (eaidova) [reply models/public/mask_rcnn_inception_v2_coco/mask_rcnn_inception_v2_coco.md]: 33 x 33 correct only for atrous models

**Comment by jkamelin** [comment]
cc @kchechil

**Comment by kchechil** [comment]
@yssemaev please double-check licenses and compare list of models against R3 plans.

**Comment by kchechil** [comment]
@Wovchena please comment if all the models are covered with some sample/demo.

File: models/public/ssdlite_mobilenet_v2/ssdlite_mobilenet_v2.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
+# ssdlite_mobilenet_v2
+
+## Use Case and High-Level Description
+
+The `ssdlite_mobilenet_v2` is used for object detection. For details, see [paper](https://arxiv.org/pdf/1801.04381.pdf) MobileNetV2: Inverted Residuals and Linear Bottlenecks.
```
Comment (IRDonch) [inline models/public/ssdlite_mobilenet_v2/ssdlite_mobilenet_v2.md]: The first sentence isn't quite grammatical. It should be one of:

* The `ssdlite_mobilenet_v2` model is...
* `ssdlite_mobilenet_v2` is...

File: tools/downloader/list_topologies.yml [hunk 9]
```diff
@@ -385,6 +385,164 @@ topologies:
     framework: caffe
     license: https://raw.githubusercontent.com/DuinoDu/mtcnn/master/LICENSE
 
+  - name: "mask_rcnn_inception_resnet_v2_atrous_coco"
+    description: >-
+      Mask R-CNN Inception Resnet V2 Atrous trained on COCO dataset. Used for object instance segmentation. For details see paper
```
Comment (IRDonch) [inline tools/downloader/list_topologies.yml]: For readability, please keep description line lengths to 80-90 characters.

File: tools/downloader/list_topologies.yml [hunk 10]
```diff
@@ -1378,6 +1536,36 @@ topologies:
...
+        format: gztar
+        file: mobilenet_v2_1.0_224.tgz
+    model_optimizer_args:
+      - --framework=tf
+      - --data_type=FP32
+      - --reverse_input_channels
+      - --input_shape=[1,224,224,3]
+      - --input=input
+      - --mean_values=input[127.5,127.5,127.5]
+      - --scale_values=input[127.00001206500114]
```
Comment (IRDonch) [inline tools/downloader/list_topologies.yml]: Why such a strange number? In the description it says just `127.0`.

Comment (jkamelin) [reply tools/downloader/list_topologies.yml]: This number was taken from benchmark output, where it is stored as 1/127.0 (and probably was rounded), and then converted to 1/(1/127.0).

**Comment by IRDonch** [comment]
Please update `tools/downloader/license.txt`.

File: models/public/mask_rcnn_inception_resnet_v2_atrous_coco/mask_rcnn_inception_resnet_v2_atrous_coco.md [hunk 11]
```diff
@@ -0,0 +1,78 @@
+# mask_rcnn_inception_resnet_v2_atrous_coco
+
+## Use Case and High-Level Description
+
+Mask R-CNN Inception Resnet V2 Atrous trained on COCO dataset. Used for object instance segmentation. For details, see [paper](https://arxiv.org/pdf/1703.06870.pdf)
```
[the remark this answers is not in the document]
Comment (IRDonch) [reply models/public/mask_rcnn_inception_resnet_v2_atrous_coco/mask_rcnn_inception_resnet_v2_atrous_coco.md]: Please update the descriptions in the configs, too.

**Pull request merged**